### PR TITLE
Adding version numbers to vsimporter and xib2nib

### PR DESCRIPTION
**Pull request by asimonov-msft**
This is my initial attempt at versioning vsimporter and xib2nib. The shared release version number has to be manually incremented and resides in build\winobjc.version.

### CODE REVIEW

File: build/winobjc.version [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#define VERSION_MAJOR               0
```
Comment (asimonov-msft) [inline build/winobjc.version]: Not quite. For now the workflow for changing the tools will be:
- Make changes
- Check in changes, including the re-built binaries, into git. Do nothing with the version.

The monthly release workflow is:
- Update the winobjc.version file with a new release string
- Build the SDK and publish the zip file.

---
In reply to: [84107821](https://github.com/Microsoft/WinObjC/pull/1184#discussion_r84107821) [](ancestors = 84107821)

File: tools/vsimporter/src/vsimporter.cpp [hunk 2]
```diff
@@ -138,19 +133,24 @@ int main(int argc, char* argv[])
 
     if (c == -1)
```
Comment (pradipd) [inline tools/vsimporter/src/vsimporter.cpp]: nit: Can you add {}

File: tools/vsimporter/src/vsimporter.cpp [hunk 3]
```diff
@@ -211,7 +211,8 @@ int main(int argc, char* argv[])
         // Due to issue 6715724, flush before exiting
 		TELEMETRY_EVENT_DATA(L"VSImporterIncomplete", "printUsage");
```
Comment (pradipd) [inline tools/vsimporter/src/vsimporter.cpp]: >		 [](start = 0, length = 2)

nit: tabs/spaces are inconsitent.

**Comment by pradipd** [comment]
:shipit:

**Comment by andrejo-msft** [comment]
:shipit:

**Comment by bbowman** [comment]
:clock1:

File: msvc/task/GenVersionFiles/RCHeaderFile.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            this.version = version;
+        }
+
+        private VersionInfo version;
+
+        public void Write(string filePath)
+        {
+            using (StreamWriter file = new System.IO.StreamWriter(filePath))
+            {
+                file.WriteLine("#define FILE_VERSION {0}, {1}, {2}, {3}", version.Major, version.Minor, version.Build, version.Revision);
```
Comment (pradipd) [inline msvc/task/GenVersionFiles/RCHeaderFile.cs]: >  file.WriteLine("#define FILE_VERSION {0}, {1}, {2}, {3 [](start = 14, length = 56)

Can you share out what the generated version files look like?

Comment (asimonov-msft) [inline msvc/task/GenVersionFiles/RCHeaderFile.cs]: SharedVersion.h:
#define FILE_VERSION 0, 2, 1610, 24
#define FILE_VERSION_STR "0.2.1610.24"

SharedAssemblyInfo.cs:
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;

[assembly: AssemblyFileVersion("0.2.1610.24")]
[assembly: AssemblyInformationalVersion("0.2.1610.24")]

---
In reply to: [84728270](https://github.com/Microsoft/WinObjC/pull/1184#discussion_r84728270) [](ancestors = 84728270)

File: msvc/tools-build.targets [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    <BuildDependsOn Condition="'$(ProjectExt)' == '.vcxproj'">GenSharedHeaderFile;$(BuildDependsOn);</BuildDependsOn>
+  </PropertyGroup>
+
+  <Target Name="MakeSharedVersionHeaderDir" >
+    <MakeDir Directories="$(SharedVersionHeaderDir)" />
+  </Target>
+
+  <Target
+    Name="GenSharedAssemblyFile"
+    Inputs="$(WinObjCVersionFile)"
```
Comment (pradipd) [inline msvc/tools-build.targets]: >WinObjCVersionFile [](start = 14, length = 18)

I'm a slightly bit concerned that each tools binary is going to rebuild the version file.  This means that each binary will always be out-of-date.
However,  given our expected use cases , I think this is acceptable.

**Pull request merged**